### PR TITLE
Optimise before swap return fee

### PR DESCRIPTION
**Pull request by hensha256**
Potential optimisation to #648

### CODE REVIEW

File: .forge-snapshots/swap skips hook call if hook is caller.snap [hunk 1]
```diff
@@ -1 +1 @@
-225454
\ No newline at end of file
+225012
```
Comment (hensha256) [inline .forge-snapshots/swap skips hook call if hook is caller.snap:1]: key gas to look at

File: .forge-snapshots/swap with lp fee and protocol fee.snap [hunk 2]
```diff
@@ -1 +1 @@
-182375
\ No newline at end of file
+181943
```
Comment (hensha256) [inline .forge-snapshots/swap with lp fee and protocol fee.snap:1]: key gas to look at

File: .forge-snapshots/update dynamic fee in before swap.snap [hunk 3]
```diff
@@ -1 +1 @@
-160612
\ No newline at end of file
+160180
```
Comment (hensha256) [inline .forge-snapshots/update dynamic fee in before swap.snap:1]: key gas to look at

File: .forge-snapshots/swap with return dynamic fee.snap [hunk 4]
```diff
@@ -1 +1 @@
-157794
\ No newline at end of file
+157355
```
Comment (hensha256) [inline .forge-snapshots/swap with return dynamic fee.snap:1]: key gas to look at

File: .forge-snapshots/swap CA custom curve + swap noop.snap [hunk 5]
```diff
@@ -1 +1 @@
-136229
\ No newline at end of file
+135805
```
Comment (hensha256) [inline .forge-snapshots/swap CA custom curve + swap noop.snap:1]: https://github.com/Uniswap/v4-core/pull/650/files

Comment (hensha256) [inline .forge-snapshots/swap CA custom curve + swap noop.snap:1]: key gas to look at

File: .forge-snapshots/swap with return dynamic fee.snap (hunk 4 above)
Comment (saucepoint) [reply .forge-snapshots/swap with return dynamic fee.snap:1]: 3,257 😮‍💨

**Review by saucepoint — APPROVED** [review]
i like it -- was one my main issues with the original impl. felt like the one-off handler `callHookWithReturnDeltaAndFee` wasnt very clean

**Pull request merged**